tachyon/crypto/commitments/merkle_tree/field_merkle_tree/field_merkle_tree.h

```diff
@@ -271,30 +271,24 @@ class FieldMerkleTree {
 
     std::vector<Digest> ret(max_rows_padded);
     absl::Span<Digest> sub_ret = absl::MakeSpan(ret).subspan(0, max_rows);
-    base::ParallelizeByChunkSize(
-        sub_ret, PackedPrimeField::N,
-        [&hasher, &packed_hasher, tallest_matrices](
-            absl::Span<Digest> chunk, size_t chunk_offset, size_t chunk_size) {
-          size_t start = chunk_offset * chunk_size;
-          if (chunk.size() == chunk_size) {
-            std::vector<PackedPrimeField> packed_prime_fields =
-                base::FlatMap(tallest_matrices, [start](RowMajorMatrixView m) {
-                  return math::PackRowVertically<PackedPrimeField>(*m, start);
-                });
-            PackedDigest packed_digest =
-                packed_hasher.Hash(packed_prime_fields);
-            for (size_t i = 0; i < chunk.size(); ++i) {
-              for (size_t j = 0; j < N; ++j) {
-                chunk[i][j] = std::move(packed_digest[j][i]);
-              }
-            }
-          } else {
-            for (size_t i = 0; i < chunk.size(); ++i) {
-              chunk[i] = hasher.Hash(
-                  GetRowAsPrimeFieldVector(tallest_matrices, start + i));
-            }
-          }
-        });
+    size_t end = (sub_ret.size() / PackedPrimeField::N) * PackedPrimeField::N;
```

Comment:
This repeated modification could be minimized by implementing `base::DynamicParallelizeByChunkSize`. Additionally, this function seems like it could be used continuously in the future.

Reply:
I've created `base::DynamicParallelizeByChunkSize`, yet thinking about it, I'm confused on why this isn't the default for `OMP_PARALLEL_FOR` since it seems like every use case of `OMP_PARALLEL_FOR` must suffer from the same static scheduler. 🤔